### ISSUE TITLE
Crash when changing georeferences

### ISSUE THREAD

**Issue by javagl** [CesiumGS/cesium-unreal#290]
Sorry of the "unspecific" description, I just want to note this here, and it has to be investigated further:

While trying out a tutorial, I made some mistake and started moving and deleting some geo references that I accidentally duplicated (I think by moving actors between sublevels). At some point, UE crashed:
```
Unhandled Exception: EXCEPTION_ACCESS_VIOLATION reading address 0x0000000000000000

UE4Editor_CesiumRuntime!ACesiumGeoreference::TransformUeToEcef() [c:\users\travis\build\cesiumgs\packages\cesiumforunreal\hostproject\plugins\cesiumforunreal\source\cesiumruntime\private\cesiumgeoreference.cpp:623]
UE4Editor_CesiumRuntime!AGlobeAwareDefaultPawn::Tick() [c:\users\travis\build\cesiumgs\packages\cesiumforunreal\hostproject\plugins\cesiumforunreal\source\cesiumruntime\private\globeawaredefaultpawn.cpp:372]
UE4Editor_Engine
UE4Editor_Engine
UE4Editor_Engine
UE4Editor_Engine
UE4Editor_Core
UE4Editor_Core
UE4Editor_Core
UE4Editor_Engine
UE4Editor_Engine
UE4Editor_Engine
UE4Editor_Engine
UE4Editor_UnrealEd
UE4Editor_UnrealEd
UE4Editor
UE4Editor
UE4Editor
UE4Editor
UE4Editor
kernel32
ntdll
```

I'll have to figure out how exactly this can be reproduced, but it likely means that at https://github.com/CesiumGS/cesium-unreal/blob/908a34fbfff2b81f32d81d4704ccb19702205519/Source/CesiumRuntime/Private/CesiumGeoreference.cpp#L622 , either the World (or less likely, the OriginLocation) had been `nullptr` for some reason.

**Comment by javagl** [comment]
To keep the record here: The issue can be reproduced reliably when deleting a georeference (and ignoring the message about the brken references). The intended behavior after deletion may not be entirely clear, but ... "it should not crash" is the baseline :-) 

According to a comment by @xuelongmu 

> Doing an `IsValid(GeoreferenceActor)` check on whatever code is invoking the GeoreferenceActor methods will ensure that the GeoreferenceActor is a) not null and b) not pending kill (aka removed from world and will be cleaned up on the next garbage collection cycle). I agree that checking `IsValid(this->GetWorld)` in the TransformUeToEcef function call is a good idea as well.

the check should *not* only be a `null` check (because that wouldn't consider pending removal), but a dedicated `IsValid` check, so I'd add that soon.

**Comment by javagl** [comment]
This is strongly enough related to mention it here: Whe setting the georeference of a tileset to `None` (the default), and pressing "Play", the UE crashes. And this crash is remarkably hard. It does not even show a crash message. It's just gone. 

The reason for that is that the "simple" solution of just adding `null`/`IsValid` checks (as in https://github.com/CesiumGS/cesium-unreal/blob/main/Source/CesiumRuntime/Private/Cesium3DTilesetRoot.cpp#L88 ) has a ripple effect: When this is done, some matrix remains `0`, and this matrix is used to transform some vector (which becomes `0`), and this vector is used as a normal vector for the view volume computation, and this creates a `Plane`, and the [constructor of `Plane` throws up in this case](https://github.com/CesiumGS/cesium-native/blob/main/CesiumGeometry/src/Plane.cpp#L13) ... 

However, the georeference is used quite a lot in the pawn and tileset classes, and I'll *try* to wrap these usages into `IsValid` checks. But there may be cases where this raises a question, and this is broadly: What to do in the `else` case? I *assume* that much of this can be solved by "just using the identity matrix", but it may not always be that simple and obvious.

**Comment by javagl** [comment]
The changes in the PR should prevent the crashes, at least insofar that deleting the georeference causes the Unreal message about broken references, but it is *still* possible to "Play" without crashing. (Of course, deleting the Georeference only on *either* one tileset *or* the pawn causes no valid rendering, but I'd say this is anticipated). 

Generally, the `Georeference` that is used in the pawn or tileset can become `null` (or invalid) ~"at any point in time", and this is now handled by checking `IsValid(this->Georeference)`, and trying to fall back to some sensible default. But there doesn't seem to be a sensible solution for the remaining usages, mainly around https://github.com/CesiumGS/cesium-unreal/blob/main/Source/CesiumRuntime/Private/GlobeAwareDefaultPawn.cpp#L343 where the georeference is used for updating the rotation during a flight.

(In doubt, one could just "end the flight" when the georeference becomes invalid - I'm not sure whether this code path can even be reached when the georeference is not valid)

---

But more broadly, while adding the fixes until now, I wondered: Should actors and pawns without a georeference work sensibly **in general**? I think this could be accomplished generically and elegantly, by just saying: If you don't have a georeference, then everything is ~"in Unreal World coordinates". This may be desirable for some cases - if somebody just wants to walk around on a tileset, but does not really care about its location on the globe - but I'm not sure whether this can be implemented consistently. Specifically, it might mean that some latitude/longitude would brutally be interpreted as x/y coordinates in some cases. And if this is a goal, I still have to wrap my head around some geosp**ec**ialities...

**Comment by kring** [comment]
Fixed in #329.
